### PR TITLE
HPCC-28987 Generate vcpkg.json 

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
Draft until https://github.com/hpcc-systems/HPCC-Platform/pull/16987 has been pulled

File: cmake_modules/plugins.cmake [hunk 1]
```diff
@@ -37,18 +39,108 @@ set(PLUGINS_LIST
...
+    set(VCPKG_LIBXSLT "${VCPKG_INCLUDE}")
+endif()
+
+set(VCPKG_LIBXALAN "${VCPKG_SUPPRESS}")
+if (USE_LIBXALAN)
+    set(VCPKG_LIBXALAN "${VCPKG_INCLUDE}")
+endif()
+
+set(VCPKG_NUMA "${VCPKG_SUPPRESS}")
+if (USE_OPENLDAP)
```
Comment (ghalliday) [inline cmake_modules/plugins.cmake:117]: USE_NUMA?

**Review by ghalliday — COMMENTED** [review]
One question

**Comment by ghalliday** [comment]
@jakesmith please sanity check.

**Review by jakesmith — APPROVED** [review]
@GordonSmith - 1 question, but looks good.

File: system/logaccess/CMakeLists.txt [hunk 2]
```diff
@@ -18,7 +18,5 @@ IF(NOT CLIENTTOOLS_ONLY)
   IF((NOT WIN32) AND USE_ELASTICSTACK_CLIENT)
     HPCC_ADD_SUBDIRECTORY (ElasticStack)
   ENDIF()
-  #IF (USE_AZURE)
-    HPCC_ADD_SUBDIRECTORY (Azure)
-  #ENDIF()
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply system/logaccess/CMakeLists.txt:23]: The Azure logaccess doesn't use the Azure libs (it just uses CURL)

File: cmake_modules/plugins.cmake (hunk 1 above)
Comment (GordonSmith) [reply cmake_modules/plugins.cmake]: Yup - resolved now...

**Review by ghalliday — APPROVED** [review]

**Pull request merged**